knotx-core/knotx-server/src/main/java/com/cognifide/knotx/server/KnotxServerVerticle.java

```diff
@@ -126,26 +126,16 @@ private JsonObject createEngineRequest(RepositoryResponse repositoryResponse, Ht
     return new TemplateEngineRequest(
         repositoryResponse.getData(),
         request.method(),
-        getPreservedHeaders(request.headers()),
+        request.headers(),
         request.params(),
         request.formAttributes(),
         request.uri())
         .toJsonObject();
   }
 
   private void rewriteHeaders(HttpServerRequest httpServerRequest, MultiMap headers) {
-    MultiMap preservedHeaders = getPreservedHeaders(headers);
-    preservedHeaders.names().forEach(headerKey -> httpServerRequest.response().putHeader(headerKey, preservedHeaders.get(headerKey)));
-  }
-
-  private MultiMap getPreservedHeaders(MultiMap headers) {
-    final MultiMap preservedHeaders = MultiMap.caseInsensitiveMultiMap();
-    headers.names().stream()
-        .filter(header -> configuration.serviceCallHeaders().contains(header))
-        .forEach(header -> preservedHeaders.add(header, headers.get(header)));
-
-    return preservedHeaders;
-  }
+        httpServerRequest.response().headers().addAll(headers);
```

Comment:
after removing all this stuff, it's going to be much more readable if you simply move code of that function directly to KnotxServerVerticle where rewriteHeaders() is currently invoked